<launch>
  <!-- use_sim_time 설정 -->
  <param name="/use_sim_time" value="false" /> <!-- rosbag 재생 시 -->

  <!-- Arguments for topics and frames -->
  <arg name="lidar_topic" default="/lidar_points"/>
  <arg name="lidar_frame" default="hesai_lidar"/>
  <arg name="imu_topic" default="/imu/data_raw"/>
  <arg name="target_frame" default="ego_car"/> <!-- Vehicle central frame -->
  <arg name="world_frame" default="world"/>     <!-- Reference world frame -->
  <arg name="camera_frame" default="gmsl_camera"/> <!-- Camera frame -->



  <node name="inspva_to_imu" pkg="lidar_tracking" type="inspva_to_imu.py" output="screen">
    <arg name="imu_topic" default="/imu/data_raw"/>
  </node>

  <!-- Cloud Segmentation 노드 -->
  <node name="cloud_segmentation" pkg="lidar_tracking" type="cloud_segmentation" output="screen">
    <rosparam command="load" file="$(find lidar_tracking)/config/patchworkpp_hesai.yaml" />
    <rosparam command="load" file="$(find lidar_tracking)/config/params.yaml" />
    <param name="lidar_topic" value="$(arg lidar_topic)"/>
    <param name="lidar_frame" value="$(arg lidar_frame)"/>
    <param name="imu_topic" value="$(arg imu_topic)"/>
    <param name="target_frame" value="$(arg target_frame)"/>
    <param name="world_frame" value="$(arg world_frame)"/>
  </node>

  <!-- Tracking 노드 -->
  <node name="tracking" pkg="lidar_tracking" type="tracking" output="screen">
    <rosparam command="load" file="$(find lidar_tracking)/config/params.yaml" />
    <param name="lidar_frame" value="$(arg lidar_frame)"/>
    <param name="target_frame" value="$(arg target_frame)"/>
    <param name="world_frame" value="$(arg world_frame)"/>
  </node>
  

  <!-- Lidar to Camera Projection 노드 추가 -->
  <node name="lidar_to_camera_projection" pkg="lidar_tracking" type="lidar_to_camera_projection" output="screen">
    <param name="lidar_topic" value="/lidar_points"/>
    <param name="lidar_frame" value="$(arg lidar_frame)"/>
    <param name="camera_frame" value="$(arg camera_frame)"/>
    <param name="world_frame" value="$(arg world_frame)"/>
    <param name="ego_frame" value="$(arg target_frame)"/> <!-- Adding ego_frame param -->
  </node>
  
  <!-- RViz 노드 (옵션) -->
  <!-- <node name="$(anon rviz)" pkg="rviz" type="rviz" args="-d $(find lidar_tracking)/rviz/hesai.rviz"/> -->
</launch>
